File: JADE/src/main/java/kislaya/Catacombs.java
package kislaya;

public class Catacombs {

    public Room first;

    public Catacombs() {
        first = new Room(null);
        Room leftRoom = new Room(first);
        Room rightRoom = new Room(first);
        first.to.add(leftRoom);
        first.to.add(rightRoom);
        Room leftLeftRoom = new Room(leftRoom);
        Room leftRightRoom = new Room(leftRoom);
        leftRoom.to.add(leftLeftRoom);
        leftRoom.to.add(leftRightRoom);
        Room rightLeftRoom = new Room(rightRoom);
        Room rightRightRoom = new Room(rightRoom);
        rightRoom.to.add(rightLeftRoom);
        rightRoom.to.add(rightRightRoom);
    }
}

File: scalaactors/src/main/scala/kislaya/Catacombs.scala
package kislaya

class Catacombs {

    var first: Room

    def Catacombs() {
        first = new Room(null)
        var leftRoom = new Room(first)
        var rightRoom = new Room(first)
        first.to.add(leftRoom)
        first.to.add(rightRoom)
        var leftLeftRoom = new Room(leftRoom)
        var leftRightRoom = new Room(leftRoom)
        leftRoom.to.add(leftLeftRoom)
        leftRoom.to.add(leftRightRoom)
        var rightLeftRoom = new Room(rightRoom)
        var rightRightRoom = new Room(rightRoom)
        rightRoom.to.add(rightLeftRoom)
        rightRoom.to.add(rightRightRoom)
    }
}
